addons/website_blog/static/src/js/blog_tag_add.js

```diff
@@ -0,0 +1,313 @@
+    },
+    /**
+     * @override
+     */
+    _renderCustomXML: async function (uiFragment) {
+        this.$tagList = $(uiFragment.querySelector('[data-name="tag_list"]'));
+        this.$select = $(uiFragment.querySelector('[data-name="add_tag_opt"] we-select'));
+        this.tagIds.forEach(tag => {
+            this.$tagList.append(qweb.render('website_blog.tagListItem', {
+                tag: tag,
+                isEditing: this.isEditing,
+            }));
+        });
+        this.availableTags.forEach(tag => {
+            this.$select.prepend(qweb.render('website_blog.tagSelectItem', {
+                tag: tag,
+                isEditing: this.isEditing,
+            }));
+        });
+    },
+    _sendUpdateData: function () {
+        this.trigger_up('set_blog_post_update_data', {
+            blogPostId: this.blogPostId,
+            addList: this.addList,
+            unlinkList: this.unlinkList,
+        });
+    },
+    /**
+     * Toggles the UI between read/edit mode
+     *
+     * @private
+     */
+    _toggleEditUI: function () {
+        const $editMenu = $('[data-name="add_tag_opt"]');
+        $editMenu.toggleClass('d-none', !this.isEditing);
+        $('.o_we_edit_tag_btn').toggleClass('d-none', this.isEditing);
+        $('.o_we_save_tag_btn').toggleClass('d-none', !this.isEditing);
+    },
+});
+
+EditorMenuBar.include({
+    custom_events: Object.assign(EditorMenuBar.prototype.custom_events, {
+        set_blog_post_update_data: '_onSetBlogPostUpdateData',
+    }),
+
+    /**
+     * @override
+     */
+    async willStart() {
+        const _super = this._super.bind(this);
+        this.addList = [];
+        this.unlinkList = [];
+        return _super(...arguments);
+    },
+    /**
+     * @override
+     */
+    async save() {
+        const _super = this._super.bind(this);
+        await _super(...arguments);
+        this._saveTags();
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Saves the tag list in the database.
+     *
+     * @private
+     */
+    _saveTags: function () {
+        if (!this.addList && !this.unlinkList) {
+            return;
+        }
+        const proms = [];
+        this.addList.forEach(tag => {
+            if (tag.id === 0) {
+                proms.push(this._rpc({
+                    model: 'blog.tag',
+                    method: 'create',
+                    args: [{
+                        'name': tag.name,
+                        'post_ids': [this.blogPostId],
+                    }],
+                }));
+            } else {
+                proms.push(this._rpc({
+                    model: 'blog.tag',
+                    method: 'write',
+                    args: [tag.id, {
+                        'post_ids': [[4, this.blogPostId, 0]],
+                    }],
+                }));
+            }
```

Comment:
Way easier to code, read and actually more rpc savings by making only: one rpc for each created tag, one rpc to replace all tags with the new ones